modules/govuk/manifests/apps/govuk_delivery.pp

```diff
@@ -1,22 +1,108 @@
-# FIXME: This class needs better documentation as per https://docs.puppetlabs.com/guides/style_guide.html#puppet-doc
+# == Class: govuk::apps::govuk_delivery
+#
+# An API to interface with govdelivery, sending email notices to users
+# subscribed to content changes on GOV.UK
+#
+# === Parameters
+#
+# [*redis_host*]
+#   Hostname of redis, used as message queue for celery
+#
+# [*redis_port*]
+#   Port of redis, used as message queue for celery
+#
+# [*mongodb_hosts*]
+#   List of hostnames running mongodb used to store the govdelivery allocated ID
+#   of a feed for later use when adding subscribers
+#
+# [*mongodb_database*]
+#   Name of the database in mongo where the govdelivery topic IDs are stored
+#
+# [*list_title_format*]
+#   String format (Python syntax) for the title of lists
+#
+# [*govdelivery_username*]
+#   Username used to authenticate with govdelivery API
+#
+# [*govdelivery_password*]
+#   Password used to authenticate with govdelivery API
+#
+# [*govdelivery_account_code*]
+#   Identifier for GOV.UK within the govdelivery service
+#
+# [*govdelivery_hostname*]
+#   Hostname for the govdelivery API (changes depending on environment)
+#
+# [*govdelivery_signup_form*]
+#   URL to the public govdelivery page users are redirected to to enter their
+#   email and subscribe (URL should include a %s to provide a topic ID)
+#
+# [*port*]
+#   Port the govuk_delivery service runs on, used in healthchecking the service.
+#
+# [*enable_procfile_worker*]
+#   Run the celery worker defined in the procfile
 class govuk::apps::govuk_delivery(
-  $enabled = true,
+  $redis_host,
+  $redis_port,
+  $mongodb_hosts,
+  $mongodb_database,
+  $list_title_format,
+  $govdelivery_username = undef,
+  $govdelivery_password = undef,
+  $govdelivery_account_code = undef,
+  $govdelivery_hostname = undef,
+  $govdelivery_signup_form = undef,
   $port = '3042',
-  $enable_procfile_worker = true,
+  $enable_procfile_worker = true
```

Comment:
Sorry for the annoying nitpick comment, please can we keep the trailing comma in here to keep future diffs smaller? Apart from that this is good to go!